telematic_system/telematic_historical_data_processing/rosbag2_processing_service/rosbag2_processing_service/service_manager.py

```diff
@@ -145,22 +146,26 @@ def create_mysql_conn(self):
 
         try:
             conn = mysql.connector.connect(user= self.config.mysql_user, password= self.config.mysql_password,
-                              host= self.config.mysql_host,
-                              database= self.config.mysql_db, port = self.config.mysql_port)
+                            host= self.config.mysql_host,
+                            database= self.config.mysql_db, port = self.config.mysql_port)
             self.config.logger.info("Connected to MySQL database!")
             return conn
         except mysql.connector.Error as err:
-            if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
+            if err.errno == mysql.connector.Error.errorcode.ER_ACCESS_DENIED_ERROR:
                 self.config.logger.error(f"Mysql User name or password not accepted for user: {self.config.mysql_user} and pass: {self.config.mysql_password}")
-            elif err.errno == errorcode.ER_BAD_DB_ERROR:
+            elif err.errno == mysql.connector.Error.errorcode.ER_BAD_DB_ERROR:
                 self.config.logger.error(f"Mysql Database {self.config.mysql_db} does not exist")
             else:
                 self.config.logger.error(f"Error connecting to mysql database: {err}")
 
 
+
     def update_mysql_entry(self, file_name, process_status, process_error_msg="NA"):
         # This method updates the mysql database entry for the rosbag to process
         # Update the update fields with update values
+        if not self.mysql_conn.is_connected():
```

Comment:
Why is there a possibility that the connection need a reconnect here?

Reply:
This is to re-establish a connection to the database, if it breaks sometime before this. This was added since we don't have any checks in between to test the connection after the service starts up